src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,140 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * A rule that checks if a fact exists between two numbers with various inclusivity options.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RangeRule extends KeyBasedRule {
+
+    private final Number min;
+    private final Number max;
+    private final boolean inclusiveMin;
+    private final boolean inclusiveMax;
+    private final Executor executor;
+
+    public RangeRule(String key, Number min, Number max) {
+        this(key, min, max, true, true); // Default to inclusive for both min and max
+    }
+
+    public RangeRule(String key, Number min, Number max, boolean inclusiveMin, boolean inclusiveMax) {
+        this(key, min, max, inclusiveMin, inclusiveMax, null);
+    }
+
+    public RangeRule(String key, Number min, Number max, boolean inclusiveMin, boolean inclusiveMax, Executor executor) {
+        super(RuleType.RANGE.getName());
+        this.setKey(key);
+        this.min = min;
+        this.max = max;
+        this.inclusiveMin = inclusiveMin;
+        this.inclusiveMax = inclusiveMax;
+        this.executor = executor;
+    }
+
+    @Override
+    public EvaluationResult evaluate(FactsResolver factsResolver) {
+        return new EvaluationResult() {
+            private Result result = Result.MAYBE;
+            private Number evaluatedFact;
+            private String message;
+            private Throwable cause;
+
+            @Override
+            public RuleResult snapshot() {
+                return ValuesRuleResult.builder().with(r -> {
+                    r.type = getType();
+                    r.key = getKey();
+                    r.description = getDescription();
+                    r.result = result;
+                    r.expectedValue = getExpectedValue();
+                    r.actualValue = evaluatedFact;
+                    r.message = message;
+                    r.cause = cause;
+                    r.ignored = isIgnore();
+                }).build();
+            }
+
+            @Override
+            public CompletableFuture<Result> status() {
+                return factsResolver.resolveFact(getKey())
+                        .thenCompose(fact -> FutureUtils.supplyAsync(() -> {
+                            if (fact instanceof Number) {
+                                Number value = (Number) fact;
+                                if (isValidRange(value)) {
+                                    result = Result.VALID;
+                                } else {
+                                    result = Result.INVALID;
+                                    message = "Value is outside the range " + getExpectedValue();
+                                }
+                                evaluatedFact = value;
+                            } else {
+                                result = Result.INVALID;
+                                message = "Fact is not a Number";
```

Comment:
I still need to run this properly, but if the fact is of the wrong type, wouldn't get get an IllegalArgumentException?

If you look at the other rules, we want to return a `Result.FAILED` on malformed data. 

A `Result.INVALID` should be used to indicate that the data is of the right format but it does not satisfy the conditions of our rule.